src/frontends/src/query_frontend_tests.cpp

```diff
@@ -110,19 +103,19 @@ TEST_F(QueryFrontendTest, TestDecodeMalformedJSON) {
   ASSERT_THROW(
       rh_.decode_and_handle_predict(gibberish_string1, "test", {},
                                     "test_policy", 30000, InputType::Doubles),
-      ptree_error);
+      json_parse_error);
```

Comment:
Replaced the `ptree_error` and `ptree_bad_data` exceptions with the `json_parse_error` and `json_semantic_error` exceptions that I've defined. This way, we don't have to rely on the JSON library's own exceptions so these tests will remain exactly the same even if we swap out JSON parsing implementations.